### DISCUSSION TITLE
A-BTN , B-BTN を リードスイッチにしたい！！

### DISCUSSION THREAD

**Discussion by Mihokun** [algyan/XIAOGYAN#71] [Q&A]
現在の A-BTN , B-BTN を リードスイッチ（すぐ壊れる形状のやつ、もしくは ホールIC）にしたいのです。

まずは、A-BTN , B-BTN が繋がる端子に、自分家の プッシュスイッチ をつないで、HardwareTest のプログラムで動作することを確認しました（図1）。

そのまんま調子にのって
プッシュスイッチ を すぐ壊れる形状のリードスイッチ に変えてみたら、
リードスイッチ と プッシュスイッチ の 反応が逆（← うまく言えない）のようで、
リードスイッチを配線しただけの状態 ＝ プッシュスイッチ押している状態
のよう。
（ホールICにしても結果は同じだった）
リードスイッチにしたいとき、なんかいい方法ってありますか？？

**Comment by Mihokun** [comment]
![図1](https://github.com/algyan/XIAOGYAN/assets/9550393/c448f74d-0c09-475e-aae4-beebe2e413d6)

.png じゃないと、写真はみれないのかな？
.jpg から .png に変更して 図1.png をここに添付！

**Reply by y-ohe** [reply]
うーん、ちょっと良く分からないですね。
XIAOGYANで使っているプッシュボタンはノーマリーオフのタイプです。
通常のリードスイッチもノーマリーオフ(磁石を離しているとき)のものがほとんどなので、動作はどちらも同じになると思われます。
リードスイッチの近くに磁石か磁石の変わりになるようなものがあるとかではないですか？
スピーカーの中には磁石が入っているので、近づけるとリードスイッチがオンになってしまうことがあると思いますよ。

**Reply by Mihokun** [reply]
わぁー！！返信もらえた！！ ^^# なんかそれだけでうれしい笑

そうなんですよ。。。
使ったリードスイッチやホールＩＣ はそれぞれ、
リードスイッチ MKA-10110
ホールIC SK8552G（ホールIC は GPIO 、 GND、3.3 につないだ）
で、リードスイッチも別のブレッドボードに挿してあるやつをそこからテストリードでつないで使ってるから10cmから15cmは離れてる。。。

自分で（長さ延長したくて）プッシュスイッチをつないだ時はちゃんと動いてるから、そのスイッチを取っ払ってリードスイッチにしたら、最初の1回、押されてるかのような処理が走るから、なにかあるんだろうなぁ。。。

**Reply by y-ohe** [reply]
> リードスイッチ MKA-10110
ホールIC SK8552G（ホールIC は GPIO 、 GND、3.3 につないだ）

これらなら、タクトスイッチ「TVGP01-G73BB with Gray cap」と同じ動きをするはずですね。
> 自分で（長さ延長したくて）プッシュスイッチをつないだ時はちゃんと動いてるから、そのスイッチを取っ払ってリードスイッチにしたら、最初の1回、押されてるかのような処理が走るから、なにかあるんだろうなぁ。。。

これって、スピーカーから「ド」の音が出るとか？
それだと、スイッチ関係なく、ソフトで「ド」の音ならしてるので、リードスイッチとかの問題じゃないです。
ソフトの方で、勝手に鳴らしていますよ。
↓
`    ////////////////////////////////////////
    // Startup Sequence

    // 「ド」音を鳴らす。
    Xiaogyan.speaker.setTone(262);  // C4
    // LEDマトリックスを全て赤色点灯。
    Xiaogyan.ledMatrix.setBrightness(2);
    Xiaogyan.ledMatrix.fillScreen(1);
    delay(200);

    // 音を止める。
    Xiaogyan.speaker.setTone(0);
    // LEDマトリックスを全て消灯。
    Xiaogyan.ledMatrix.fillScreen(0);`

**Reply by Mihokun** [reply]
いけたー！！！

大栄さんの「リードスイッチの近くに磁石か磁石の変わりになるようなものがあるとかではないですか？
スピーカーの中には磁石が入っているので、近づけるとリードスイッチがオンになってしまうことがあると思いますよ。」

から、スピーカーは近くになかったけど、不精（ブレボの写真）して 以前サンプルプログラムを動かした後、スイッチがほしいと、プッシュスイッチを配線に加えうまく行った のち リードスイッチを配線に加えうまく行った その ブレッドボード に置いた リードスイッチを使ったの。

大栄さんの言葉を聞いた後、ちょっと休もうとシャワして ^^ そして、そうだ、あのブレッドボードで何か起きているかもしれない！と。
シンプルに リードスイッチだけつないで実行しよう！と、やってみた。
スイッチは もう一個くらいほしいから、D10を使おうと、そこにリードスイッチをつなげた（XIAOGYAN基板の写真）

ありがとう！！！ 大栄さんの言葉で素に戻ろう！と思えたよ！！^^#
![XIAOBLE](https://github.com/algyan/XIAOGYAN/assets/9550393/e73c09de-eda1-464f-b49a-0f38fcdf98d9)
![リードスイッチ_D10](https://github.com/algyan/XIAOGYAN/assets/9550393/c26d60b9-f045-44e8-8e1a-97c190d3fd12)

**Reply by Mihokun** [reply]
だから、きっと、Button-A も Button-B も リードスイッチにしても動くはずや ^^#
この根本の銀色から線出せば行けるよね？ ^^
![根元](https://github.com/algyan/XIAOGYAN/assets/9550393/b1edec86-6d0f-443f-b54f-3016fe18fde9)

**Comment by matsujirushi** [comment]
@Mihokun 
「いけたー！！！」とありますが、[最初の質問](https://github.com/algyan/XIAOGYAN/discussions/71#discussion-5820617)に対する有効な回答はどれでしょうか？（**Mark as answer**を付けたい。）

分かりやすいポストがなければ、、、新たに整理して説明する文書をポストすると、あとから見た人が迷わなくて良いと思います。
